packages/editor-ui/src/components/WorkflowSelectorParameterInput/WorkflowSelectorParameterInput.vue

```diff
@@ -0,0 +1,326 @@
+	parameterIssues: () => [],
+});
+
+const emit = defineEmits<{
+	'update:modelValue': [value: INodeParameterResourceLocator];
+	drop: [data: string];
+	modalOpenerClick: [];
+	focus: [];
+	blur: [];
+}>();
+
+const router = useRouter();
+const workflowsStore = useWorkflowsStore();
+const i18n = useI18n();
+const container = ref<HTMLDivElement>();
+const dropdown = ref<ComponentInstance<typeof ResourceLocatorDropdown>>();
+
+const width = ref(0);
+const inputRef = ref<HTMLInputElement | undefined>();
+
+const { isListMode, getUpdatedModePayload, selectedMode, supportedModes, getModeLabel } =
+	useWorkflowResourceLocatorModes(
+		computed(() => props.modelValue),
+		router,
+	);
+const { hideDropdown, isDropdownVisible, showDropdown } = useWorkflowResourceLocatorDropdown(
+	isListMode,
+	inputRef,
+);
+
+const {
+	hasMoreWorkflowsToLoad,
+	isLoadingResources,
+	filteredResources,
+	onSearchFilter,
+	searchFilter,
+	getWorkflowName,
+	populateNextWorkflowsPage,
+	setWorkflowsResources,
+} = useWorkflowResourcesLocator(router);
+
+const valueToDisplay = computed<NodeParameterValue>(() => {
+	if (typeof props.modelValue !== 'object') {
+		return props.modelValue;
+	}
+
+	if (isListMode.value) {
+		return props.modelValue ? props.modelValue.cachedResultName ?? props.modelValue.value : '';
+	}
+
+	return props.modelValue ? props.modelValue.value : '';
+});
+
+const placeholder = computed(() => {
+	if (isListMode.value) {
+		return i18n.baseText('resourceLocator.mode.list.placeholder');
+	}
+
+	return i18n.baseText('resourceLocator.id.placeholder');
+});
+
+function setWidth() {
+	const containerRef = container.value as HTMLElement | undefined;
+	if (containerRef) {
+		width.value = containerRef?.offsetWidth;
+	}
+}
+
+function getWorkflowUrl(workflowId: string) {
+	const { href } = router.resolve({ name: VIEWS.WORKFLOW, params: { name: workflowId } });
+	return href;
+}
+
+function onInputChange(value: string): void {
+	const params: INodeParameterResourceLocator = { __rl: true, value, mode: selectedMode.value };
+	if (isListMode.value) {
+		const resource = workflowsStore.getWorkflowById(value);
+		if (resource?.name) {
+			params.cachedResultName = getWorkflowName(value);
+		}
+	}
+	emit('update:modelValue', params);
+}
+
+function onListItemSelected(value: string) {
+	onInputChange(value);
+	hideDropdown();
+}
+
+function onInputFocus(): void {
+	setWidth();
+	showDropdown();
+	emit('focus');
+}
+
+function onInputBlur(): void {
+	emit('blur');
+}
+
+async function onDrop(data: string) {
+	emit('drop', data);
+}
+
+function onModeSwitched(mode: ResourceLocatorModes) {
+	emit('update:modelValue', getUpdatedModePayload(mode));
+}
+function onKeyDown(e: KeyboardEvent) {
+	if (isDropdownVisible.value) {
+		props.eventBus.emit('keyDown', e);
+	}
+}
+
+function openWorkflow() {
+	window.open(getWorkflowUrl(props.modelValue.value?.toString() ?? ''), '_blank');
+}
+
+onMounted(() => {
+	props.eventBus.on('searchFilter', onSearchFilter);
```

Comment:
not sure what this one does -- there doesn't seem to be a `eventBus.emit('searchFilter')` somewhere...

Reply:
same for the dropdownHide?